### DISCUSSION TITLE
我尝试将SVRT中的解码方式改为attention解码，在训练时acc正常，但是我用tools/infer_rec.py进行推理时，识别结果都是‘nan’

### DISCUSSION THREAD

**Discussion by Dhao007** [PaddlePaddle/PaddleOCR#13336] [Q&A]
配置文件如下：
Global:
  use_gpu: true
  epoch_num: 50
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./experiment/icdar2023_seal_clean/svtr_alldata_attention2/
  save_epoch_step: 1000
  eval_batch_step:
  - 0
  - 1000
  cal_metric_during_train: true
  pretrained_model: ./pretrain_models/rec/rec_svtr_tiny_none_ctc_ch_train/best_accuracy.pdparams
  checkpoints: null
  save_inference_dir: null
  use_visualdl: true
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ./train_data/icdar2023rest/official_dataset/chars.txt
  max_text_length: 40
  infer_mode: false
  use_space_char: false
  save_res_path: ./output/rec/predicts_svtr_tiny_ch_all.txt
  d2s_train_image_shape:
  - 3
  - 32
  - 320
Optimizer:
  name: AdamW
  beta1: 0.9
  beta2: 0.99
  epsilon: 1.0e-08
  weight_decay: 0.05
  no_weight_decay_name: norm pos_embed
  one_dim_param_no_weight_decay: true
  lr:
    name: Cosine
    learning_rate: 0.0005
    warmup_epoch: 2
Architecture:
  model_type: rec
  algorithm: SVTR
  Transform: null
  Backbone:
    name: SVTRNet
    img_size:
    - 32
    - 320
    out_char_num: 40
    out_channels: 96
    patch_merging: Conv
    embed_dim:
    - 64
    - 128
    - 256
    depth:
    - 3
    - 6
    - 3
    num_heads:
    - 2
    - 4
    - 8
    mixer:
    - Local
    - Local
    - Local
    - Local
    - Local
    - Local
    - Global
    - Global
    - Global
    - Global
    - Global
    - Global
    local_mixer:
    - - 7
      - 11
    - - 7
      - 11
    - - 7
      - 11
    last_stage: true
    prenorm: false
  Neck:
    name: SequenceEncoder
    encoder_type: reshape
  Head:
    name: AttentionHead
    hidden_size: 96
Loss:
  name: AttentionLoss
PostProcess:
  name: AttnLabelDecode
Metric:
  name: RecMetric
  main_indicator: acc
Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2023_seal_clean
    label_file_list:
    - ./train_data/icdar2023_seal_clean/icdar2023_all_train.txt
    - ./train_data/icdar2023_seal_clean/synthetic_data_A_train_chn.txt
    - ./train_data/icdar2023_seal_clean/synthetic_data_B_train_chn.txt
    ratio_list:
    - 1.0
    - 0.5
    - 0.5
    ext_op_transform_idx: 1
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecConAug:
        prob: 0.5
        ext_data_num: 2
        image_shape:
        - 32
        - 320
        - 3
    - SVTRRecAug:
        aug_type: 0
    - RecAug: null
    - AttnLabelEncode: null
    - SVTRRecResizeImg:
        image_shape:
        - 3
        - 32
        - 320
        padding: true
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: true
    batch_size_per_card: 64
    drop_last: false
    num_workers: 4
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2023_seal_clean
    label_file_list:
    - ./train_data/icdar2023_seal_clean/A_val.txt
    - ./train_data/icdar2023_seal_clean/B_val.txt
    - ./train_data/icdar2023_seal_clean/other_val.txt
    - ./train_data/icdar2023_seal_clean/same1_val.txt
    - ./train_data/icdar2023_seal_clean/same2_val.txt
    - ./train_data/icdar2023_seal_clean/same3_val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - AttnLabelEncode: null
    - SVTRRecResizeImg:
        image_shape:
        - 3
        - 32
        - 320
        padding: true
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 64
    num_workers: 2
profiler_options: null
